File: obi/2021/fase3/cpp/casamento.cpp
//A e B terem o mesmo numero de digitos
//o digito de menor valor da posicao i eh eliminado, se = nenhum eh eliminado

#include <bits/stdc++.h>

using namespace std;

int main(){
  string n1, n2;
  cin >> n1 >> n2;
  while (n1.size() > n2.size()) n2 = '0' + n2;
  while (n1.size() < n2.size()) n1 = '0' + n1;

  string casado1, casado2;
  for(int i = 0; i < n1.size(); i++){
    if(n1[i] == n2[i]){
      casado1 += n1[i];
      casado2 += n2[i];
    }
    else if(n1[i] > n2[i]) casado1 += n1[i];
    else casado2 += n2[i];
  }
  int res1, res2;
  if(casado1.size() == 0) res1 = -1;
  else res1 = stoi(casado1);
  if(casado2.size() == 0) res2 = -1;
  else res2 = stoi(casado2);
  cout << min(res1, res2) << " " << max(res1, res2);
}

File: exercises/anteriores/2021/fase3/casamento.py
#fazemos A e B terem o mesmo numero de digitos adicionando zero a esquerda
#cada digito de A eh comparado com o correspondente de B, e de menor valor eh eliminado
#se forem iguais mantem
#caso nao haja nenhum restante, retornar -1

a = int(input())
b = int(input())

def casamento(a,b):
  while len(a) != len(b):
    if len(a) > len(b):
      temp = b[0]
      b[0] = 0
      for i in range(len(b)):
        b[i] = b[i+1]
      b[0] = 0
    if len(a) < len(b):
      numeros[2].insert(0,0)
  return(numeros)

print(casamento(numeros))